### ISSUE TITLE
Issue with reserved words in table definitions

### ISSUE THREAD

**Issue by aadant** [yandex/ClickHouse#4921]
create table reserved_word_table(\`index\` UInt8) engine=MergeTree order by \`index\`;

CREATE TABLE reserved_word_table
(
    \`index\` UInt8
)
ENGINE = MergeTree
ORDER BY \`index\`

Ok.

0 rows in set. Elapsed: 0.009 sec.

:) alter table reserved_word_table add column c2 UInt8;

ALTER TABLE reserved_word_table
    ADD COLUMN
    c2 UInt8

Received exception from server (version 19.4.1):
Code: 62. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: Syntax error (in file /var/lib/mysql/clickhouse/metadata/test//reserved_word_table.sql): failed at position 52 (line 4, col 1): )
ENGINE = MergeTree
ORDER BY index
SETTINGS index_granularity = 8192
. Expected one of: NOT, CASE, LEFT, RIGHT, case, logical-NOT expression, logical-AND expression, logical-OR expression, lambda expression, parenthesized expression, SELECT subquery, compound identifier, identifier, qualified asterisk, asterisk, CAST expression, SUBSTRING expression, TRIM expression, LEFT expression, RIGHT expression, EXTRACT expression, DATE_ADD expression, DATE_DIFF expression, element of expression, SUBSTRING, EXTRACT, DATEADD, DATE_ADD, TIMESTAMPADD, TIMESTAMP_ADD, DATESUB, DATE_SUB, TIMESTAMPSUB, TIMESTAMP_SUB, INTERVAL, LTRIM, TRIM, RTRIM, DATEDIFF, DATE_DIFF, TIMESTAMPDIFF, TIMESTAMP_DIFF, list of elements, list, delimited by binary operators, list, delimited by operator of variable arity, expression with prefix unary operator, tuple element expression, array element expression, unary minus expression, multiplicative expression, INTERVAL operator expression, additive expression, string concatenation expression, BETWEEN expression, comparison expression, nullity checking, expression with ternary operator, OpeningRoundBracket, Arrow, array, token, string literal, literal, CAST, NULL, number, function.

0 rows in set. Elapsed: 0.001 sec.

**Comment by aadant** [comment]
Incidently, the clickhouse will not restart after creating such a table ...

**Comment by alexey-milovidov** [comment]
The issue has been fixed in #4782

**Comment by alexey-milovidov** [comment]
The fix is only available in v19.5.1.246-testing.
It's currently in release candidate stage.
You can install this version from testing repository.

```
echo "deb http://repo.yandex.ru/clickhouse/deb/testing/ main/" | sudo tee /etc/apt/sources.list.d/clickhouse.list
```
